datanommer.models/datanommer/models/__init__.py

```diff
@@ -128,7 +131,14 @@ def add(envelope):
     obj.msg = message['msg']
     obj.headers = headers
 
-    session.add(obj)
+    try:
+        session.add(obj)
+        session.flush()
```

Comment:
Did you mean ``session.commit`` here?

``flush`` just pushes everything to the database's transaction buffer and doesn't actually commit the transaction. You might get an ``IntegrityError`` if the message was already committed to the database before the transaction began, but you definitely won't see one if there's a concurrent transaction that's also adding this message. Whoever commits first wins, but the call to commit later isn't handling the conflict.